test/aggregation-builder.js

```diff
@@ -0,0 +1,86 @@
+import test from 'tape'
+import aggregationBuilder from '../src/aggregation-builder'
+import filterBuilder from '../src/filter-builder'
+
+test('aggregationBuilder | avg aggregation', (t) => {
+  t.plan(1)
+
+  const result = aggregationBuilder().aggregation('avg', 'grade')
+
+  t.deepEqual(result.getAggregations(), {
+    agg_avg_grade: {
+      avg: {
+        field: 'grade'
+      }
+    }
+  })
+})
+
+test('aggregationBuilder | percentiles aggregation', (t) => {
+  t.plan(1)
+
+  const result = aggregationBuilder().aggregation('percentiles', 'load_time', {
+    percents: [95, 99, 99.9]
+  })
+
+  t.deepEqual(result.getAggregations(), {
+    agg_percentiles_load_time: {
+      percentiles: {
+        field: 'load_time',
+        percents: [95, 99, 99.9]
+      }
+    }
+  })
+})
+
+// Skipping, first need a way to handle aggregations with no `field`
+test.skip('aggregationBuilder | percentiles script aggregation', (t) => {
+  t.plan(1)
+
+  const result = aggregationBuilder().aggregation('percentiles', 'load_time', {
+    script: {
+      inline: "doc['load_time'].value / timeUnit",
+      params: {
+        timeUnit: 100
+      }
+    }
+  })
+
+  t.deepEqual(result.getAggregations(), {
+    agg_percentiles_load_time: {
+      percentiles: {
+        script: {
+          inline: "doc['load_time'].value / timeUnit",
+          params: {
+            timeUnit: 100
+          }
+        }
+      }
+    }
+  })
+})
+
+test('aggregationBuilder | filters aggregation', (t) => {
```

Comment:
@elliott-davis FYI this test shows what I had in mind for building filters aggregations in the next version of bodybuilder. To me it feels more natural than passing a callback argument and easier to understand the usage.